### ISSUE TITLE
[BUG] Microsoft.ServiceBus vs Microsoft.Azure.ServiceBus

### ISSUE THREAD

**Issue by romankiss** [Azure/azure-sdk-for-net#15399]
This is an issue reported to https://github.com/Azure/azure-service-bus/issues/373

I have a simple scenario where the message sent on the topic will be forwarded to the queue as a scheduled message.  For this behavior the topic subscription requires to setup the following properties:

 - ForwardTo
         `name of the queue entity`

- $Default Rule
    Filter:
            ` 1=1 `
    Action (example for 10 minutes):

      SET sys.TimeToLive = '00:10:00';
      SET EnqueuedTimeUtc = sys.EnqueuedTimeUtc;
      SET ScheduledEnqueueTimeUtc = sys.ExpiresAtUtc;
      SET sys.ScheduledEnqueueTimeUtc = sys.ExpiresAtUtc;
      SET sys.Label = 'Delay';
      SET sys.TimeToLive = '01:00:00';

              
The above scenario is suitable for publishers without setting the _ScheduledEnqueueTimeUtc_ property, for instance an Azure Event Grid, see the following [usage](https://stackoverflow.com/questions/63933101/how-to-postponed-event-trigger-receive). 

For test purposes, I have used for publishing event on the topic the following:

1. tool Service Bus Explorer (the latest release such as 4.1.112) - using the Microsoft.ServiceBus.dll
2. send message using the REST POST request
3. Azure portal, Service Bus Explorer (preview) (probably underlaying logic is using the REST APIs) 
4. Azure Stream Analytics Job - output to Service Bus Topic (probably underlaying logic is using the REST APIs) 
5. using a BrokeredMessage and Microsoft.ServiceBus.dll

All the above publishers sent the message on the topic and the message is forwarding to the queue as a scheduled message based on the rule action. That's a correct expectable behavior.  

What is the issue?

The issue is when the publisher using the new assembly such as **Microsoft.Azure.ServiceBus.dll**, so the _Azure Event Grid_, _Azure Function_ with ServiceBus output binding, etc. will not correct work at the copy of the message for ForwardTo process.
The reason is that the value of the _ScheduledEnqueueTimeUtc_ is always populated by value 1/1/0001 12:00:00 AM.

My testing shown that the rule action is working well such as setting the _ScheduledEnqueueTimeUtc_ when the ForwardTo property is empty. Once the ForwardTo property is configured, the _ScheduledEnqueueTimeUtc_ is overwritten by default value (1/1/0001 12:00:00 AM).

Based on the above, it looks like this is a bug and any publishers who is using the _Microsoft.Azure.ServiceBus.dll_ assembly will have this issue. 

Note, there is no workaround for this issue for that publishers where their integration is without customizing the system properties of the service bus message, for instance: Azure Event Grid. 

Thanks
Roman

**Comment by jsquire** [comment]
//cc: @JoshLove-msft, @MiYanni

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by axisc** [comment]
@romankiss thanks for raising this issue. We will investigate this potential bug and update the thread.

**Comment by yvgopal** [comment]
I have just reproduced this behavior.
1. It is not a client SDK issue, per se. The setup works with messages sent using SBMP/Http/NetMessaging. But it doesn't work for messages sent with AMQP. Even with Microsoft.ServiceBus.dll, if you change the transport type (add TransportType=Amqp to the connection string), it won't work. The new SDKs are AMQP only and so it fails.
2. It assumes some internals of the service which are not documented. We never documented that we set one property to the message updating its dependent properties and then another property. This rule action assumes that setting TimeToLive property is immediately reflected (in the service) in ExpiresAtUtc property to be accessed in the next statement. I think these assumptions are not reliable and can stop working anytime.

Having said these two points, I can't promise I will fix it. I will have to talk to my team and see if we want to do this. Also I am not sure if this can be considered a bug, as it never worked with AMQP with any SDK.
